nixos/modules/services/web-apps/funkwhale/funkwhale.nix

```diff
@@ -0,0 +1,329 @@
+          after = [ "redis.service" "postgresql.service" ];
+          wantedBy = [ "funkwhale-init.service" ];
+          before   = [ "funkwhale-init.service" ];
+          serviceConfig = {
+            User = "postgres";
+          };
+          script = ''
+            ${pkgs.postgresql}/bin/psql -d funkwhale -c 'CREATE EXTENSION IF NOT EXISTS "unaccent";'
+          '';
+        };
+        funkwhale-init = {
+          description = "Funkwhale initialization";
+          after = [ "funkwhale-psql-init.service" ];
+          wantedBy = [ "funkwhale-server.service" "funkwhale-worker.service" "funkwhale-beat.service" ];
+          before   = [ "funkwhale-server.service" "funkwhale-worker.service" "funkwhale-beat.service" ];
+          environment = funkwhaleEnv;
+          script = ''
+            chmod a+rx ${funkwhaleHome}
+            if ! test -e ${cfg.api.media_root}; then
+            mkdir -p ${cfg.api.media_root}
+            mkdir -p ${cfg.api.static_root}
+            mkdir -p ${cfg.music_directory_path}
+            echo "#!/bin/sh
+            
+            ${pythonEnv}/bin/python ${pkgs.funkwhale}/manage.py createsuperuser" > ${funkwhaleHome}/createSuperUser.sh
+            chmod u+x ${funkwhaleHome}/createSuperUser.sh
+            chown -R funkwhale.users ${funkwhaleHome}
+
+            ${pythonEnv}/bin/python ${pkgs.funkwhale}/manage.py migrate
+            ${pythonEnv}/bin/python ${pkgs.funkwhale}/manage.py collectstatic
+            fi
+            if ! test -e ${funkwhaleHome}/config; then
+              mkdir -p ${funkwhaleHome}/config
+              ln -s ${funkwhaleEnvFile} ${funkwhaleHome}/config/.env
+              ln -s ${funkwhaleEnvFile} ${funkwhaleHome}/.env
+            fi
+          '';
+        };
+
+        funkwhale-server = {
+          description = "Funkwhale application server";
+          partOf = [ "funkwhale.target" ];
+
+          serviceConfig = serviceConfig;
+          environment = funkwhaleEnv;
+          script = "${pythonEnv}/bin/daphne -b ${cfg.api_ip} -p ${cfg.api_port} config.asgi:application --proxy-headers";
+
+          wantedBy = [ "multi-user.target" ];
+        };
+
+        funkwhale-worker = {
+          description = "Funkwhale celery worker";
+          partOf = [ "funkwhale.target" ];
+
+          serviceConfig = serviceConfig;
+          environment = funkwhaleEnv;
+          script = ''
+            if ! test -e ${funkwhaleHome}/run; then
+            mkdir -p ${funkwhaleHome}/run
+            fi
+            ${pythonEnv}/bin/celery -A funkwhale_api.taskapp worker -l INFO
+            '';
+
+          wantedBy = [ "multi-user.target" ];
+        };
+
+        funkwhale-beat = {
+          description = "Funkwhale celery beat process";
+          partOf = [ "funkwhale.target" ];
+
+          serviceConfig = serviceConfig;
+          environment = funkwhaleEnv;
+          script = ''
+            if ! test -e ${funkwhaleHome}/run; then
+            mkdir -p ${funkwhaleHome}/run
```

Comment:
Check out systemd's `RuntimeDirectory` option, no need to do this manually.

Reply:
Thanks. It's fixed